Pod/Classes/GraphQL/ARGraphQLQueryCache.m

```diff
@@ -61,7 +107,13 @@ - (instancetype)init;
 
 static NSString *
 CacheKey(NSString *queryID, NSDictionary *variables) {
-    NSData *variablesJSONData = [NSJSONSerialization dataWithJSONObject:variables options:0 error:nil];
+    // THe variables dictionary may come from anywhere and that may influence the order in many ways,
+    // as such we apply some sorting to generate a stable key.
+    NSArray *sortedKeys = [[variables allKeys] sortedArrayUsingSelector:@selector(compare:)];
+    NSArray *sortedValues = [variables objectsForKeys:sortedKeys notFoundMarker:@""];
+    NSArray *sortedVariables = [sortedKeys arrayByAddingObjectsFromArray:sortedValues];
+
+    NSData *variablesJSONData = [NSJSONSerialization dataWithJSONObject:sortedVariables options:0 error:nil];
```

Comment:
This drove me mad for a good 15 minutes 😄

Reply:
I bet!